### PR TITLE
Use https + fix some broken links + reduce redirects

### PR DESCRIPTION
**Pull request by ST-DDT**
Use HTTPS Links where possible.

Fixes https://github.com/SpongePowered/SpongeDocs/issues/437#issuecomment-338653022 (partially)

### CODE REVIEW

File: source/about/glossary.rst [hunk 1]
```diff
@@ -78,7 +78,7 @@ Sponge Forums
   regularly. https://forums.spongepowered.org/
 Sponge Foundation
   The financial side of the project, a separate entity that accepts, manages and distributes donations.
-  http://spongefoundation.org/
+  https://spongefoundation.org/
```
Comment (Minecrell) [inline source/about/glossary.rst]: We can remove this link entirely, it just links to our main homepage.

**Review by stephan-gh — CHANGES_REQUESTED** [review]

**Comment by Spongy** [comment]
Setting up PR reference, please wait...

File: source/server/getting-started/implementations/spongeforge.rst [hunk 2]
```diff
@@ -92,7 +92,7 @@ Dedicated Servers
 Installing Forge via Commandline
 --------------------------------
 
-1. Visit the `Minecraft Forge website <https://files.minecraftforge.net/>`_ and click "Show all downloads" to view the full
+1. Visit the `Minecraft Forge website <http://files.minecraftforge.net/>`_ and click "Show all downloads" to view the full
```
Comment (Minecrell) [inline source/server/getting-started/implementations/spongeforge.rst]: Try using two underscores here (and below). It's a strange warning and I've no idea why it only happens with https, but that seems to fix it according to https://github.com/sphinx-doc/sphinx/issues/3921

**Review by stephan-gh — APPROVED** [review]

**Comment by ST-DDT** [comment]
Note to self -> ToDo List:

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/about/glossary.rst#L22 | http://www.minecraftforge.net/wiki/Core_Mod is dead. 

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/about/glossary.rst#L32  | http://www.minecraftforge.net/wiki/ -> https://mcforge.readthedocs.io/en/latest/ 

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/about/history.rst#L63 | http://dragonspainter.deviantart.com/ -> https://dragonspainter.deviantart.com/

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/conf.py#L94 | http://zml2008.github.io/configurate/apidocs/ -> http://configurate.aoeu.xyz/apidocs/

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/plugin/blocks/concepts.rst#L60 | http://minecraft.gamepedia.com/NBT_format -> https://minecraft.gamepedia.com/NBT_format

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/plugin/text/representations/formatting-code-legacy.rst#L15 | http://minecraft.gamepedia.com/Formatting_codes -> https://minecraft.gamepedia.com/Formatting_codes

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/plugin/text/representations/json.rst#L7 | http://minecraft.gamepedia.com/Commands#Raw_JSON_Text -> https://minecraft.gamepedia.com/Commands#Raw_JSON_Text

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/plugin/text/text.rst#L142 | http://minecraft.gamepedia.com/Commands#Target_selectors -> https://minecraft.gamepedia.com/Commands#Target_selectors

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/server/getting-started/configuration/server-properties.rst#L57 | http://minecraft.gamepedia.com -> https://minecraft.gamepedia.com/Minecraft_Wiki

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/server/getting-started/configuration/server-properties.rst#L137 | http://minecraft.gamepedia.com/Superflat -> https://minecraft.gamepedia.com/Superflat

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/server/getting-started/configuration/server-properties.rst#L139 | http://minecraft.gamepedia.com/Customized -> https://minecraft.gamepedia.com/Customized

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/server/getting-started/implementations/spongeforge.rst#L5 | http://www.minecraftforge.net/ -> http://www.minecraftforge.net/forum/ 

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/server/getting-started/implementations/spongeforge.rst#L71 | http://files.minecraftforge.net/ -> https://files.minecraftforge.net/

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/server/management/bans.rst#L19 | http://minecraft.gamepedia.com/Commands#ban -> https://minecraft.gamepedia.com/Commands/ban

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/server/management/permissions.rst#L14 | http://minecraft.gamepedia.com/Op -> https://minecraft.gamepedia.com/Op

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/server/management/permissions.rst#L21 | http://minecraft.gamepedia.com/Commands#Summary_of_commands -> https://minecraft.gamepedia.com/Commands#Summary_of_commands

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/server/spongineer/bugreport.rst#L18 | http://pastebin.com/ -> https://pastebin.com/

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/server/spongineer/commands.rst#L201 | http://minecraft.gamepedia.com/Commands#List_of_commands -> https://minecraft.gamepedia.com/Commands#List_of_commands

EDIT:

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/plugin/plugin-meta.rst#L91 | start/project/gradle -> https://github.com/SpongePowered/SpongeGradle

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/plugin/effects.rst#L74 | `:javadoc:BlockParticle ` dead link

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/plugin/event/listeners.rst#L184 | `:javadoc:SpongeEventFactory` dead 

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/plugin/data/custom/datamanipulators.rst#L36 | `:javadoc:ItemStack` dead

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/plugin/data/custom/datamanipulators.rst#L39 | `:javadoc:BoundedComparableValue` dead

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/plugin/data/custom/datamanipulators.rst#L40 | _and lower bound of an_ -> _and lower bound of a_

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/plugin/data/custom/datamanipulators.rst#L40 | `:javadoc:Comparable` dead

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/plugin/data/custom/serialization.rst#L21 | `:javadoc:DataSerializable` dead

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/plugin/data/custom/serialization.rst#L24 | `:javadoc:DataManipulatorBuilder` dead 

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/plugin/data/custom/serialization.rst#L59 | `:javadoc:DataSerializable` dead

EDIT 2:

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/index.rst#L32 | https://spongepowered.org/downloads -> https://www.spongepowered.org/downloads

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/index.rst#L12 | https://spongepowered.org/downloads -> https://www.spongepowered.org/downloads

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/index.rst#L124 |
https://spongepowered.org -> https://www.spongepowered.org/

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/index.rst#L127 | https://www.spongepowered.org/ -> https://www.spongepowered.org/

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/index.rst#L131 | https://reddit.com/r/SpongeProject -> https://www.reddit.com/r/spongeproject/

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/index.rst#L133 | https://www.patreon.com/Sponge?ty=h -> https://www.patreon.com/Sponge

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/server/getting-started/implementations/spongeforge.rst#L13 | https://spongepowered.org/downloads -> https://www.spongepowered.org/downloads

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/server/getting-started/implementations/spongevanilla.rst#L20 | https://spongepowered.org/downloads -> https://www.spongepowered.org/downloads

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/preparing/ide.rst#L18 | https://www.jetbrains.com/idea/download -> https://www.jetbrains.com/idea/download/

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/preparing/text.rst#L5 | http://sphinx-doc.org/rest.html -> http://www.sphinx-doc.org/en/fix/https/rest.html

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/plugin/buildsystem.rst#L73 | https://repo.spongepowered.org/maven/org/spongepowered/spongeapi/ -> https://www.spongepowered.org/downloads/spongeapi/

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/about/glossary.rst#L57 | https://www.patreon.com/Sponge?ty=h -> https://www.patreon.com/Sponge

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/plugin/workspace/idea.rst#L84 | https://www.jetbrains.com/idea/help/using-git-integration.html -> https://www.jetbrains.com/help/idea/using-git-integration.html

- [ ] https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/contributing/implementation/codestyle.rst#L71 | https://code.google.com/p/guava-libraries/wiki/PreconditionsExplained -> https://github.com/google/guava/wiki/PreconditionsExplained


There are 10/20 issues like these, they should be solved by updating the link of the javadoc in conf.py:

- https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/plugin/injection.rst#L29 | `:javadoc:ConfigurationLoader<CommentedConfigurationNode>` points to http://zml2008.github.io/configurate/apidocs/ninja/leaping/configurate/loader/ConfigurationLoader.html which redirects to http://configurate.aoeu.xyz/apidocs/ninja/leaping/configurate/loader/ConfigurationLoader.html

- https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/plugin/injection.rst#L50 | same as above but for `:javadoc:GuiceObjectMapperFactory`

- https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/plugin/configuration/index.rst#L111 | same as above but for `:javadoc:YAMLConfigurationLoader`

- https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/plugin/configuration/index.rst#L112 | same as above but for `:javadoc:HoconConfigurationLoader`

- https://github.com/SpongePowered/SpongeDocs/blame/fix/https/source/plugin/text/representations/configurate.rst#L53 | same as above.

**Comment by ST-DDT** [comment]
@ImMorpheus I hope i fixed all your findings correctly. How did you find them? Did you use a tool to check for them?

@Inscrutable Any idea how i should resolve the remaining entries? (Under `Not Fixed yet`) Or who might know how to properly update the references?

File: source/server/management/bans.rst [hunk 3]
```diff
@@ -16,4 +16,4 @@ list of banned IP addresses is available using the command ``/banlist ips``
 A ban can be reversed using the command ``/pardon <name>`` or ``/pardon <ip-address>``
 
 
-More information on Bans can be found at the `Minecraft Wiki <http://minecraft.gamepedia.com/Commands#ban>`_.
+More information on Bans can be found at the `Minecraft Wiki <https://minecraft.gamepedia.com/Commands#ban>`_.
```
Comment (ImMorpheus) [inline source/server/management/bans.rst]: Should be https://minecraft.gamepedia.com/Commands/ban

File: source/server/spongineer/bugreport.rst [hunk 4]
```diff
@@ -15,7 +15,7 @@ Whenever you report a bug, please include the following:
 * any other mods or plugins installed including their versions
 * your :doc:`logs`
 
- * please use a service like `GitHubs gist <https://gist.github.com/>`_ or `pastebin <http://pastebin.com/>`_ to provide
+ * please use a service like `GitHubs gist <https://gist.github.com/>`_ or `pastebin <https://pastebin.com/>`_ to provide
```
Comment (ImMorpheus) [inline source/server/spongineer/bugreport.rst]: The SpongeForge issue template discourage the use of pastebin/hastebin. See: https://github.com/SpongePowered/SpongeForge/commit/488af7ca45fa054eea7ba066ed12d401a2ab5bca 
This should probably be updated to reflect those changes.

**Comment by ImMorpheus** [comment]
@ST-DDT Yup, good job. 
And yes, https://github.com/wummel/linkchecker.

**Comment by Inscrutable** [comment]
Awesome work! I can only really comment on the CoreMod definition - we could just add a description of what a Coremod *was* (I'll write it up if needed). I have no idea if Patreon has secret parameters, you might need to ask Dockter. You can safely edit them out for now anyway.

**Comment by ST-DDT** [comment]
@Inscrutable If you could provide me with the CoreMod definition that would be nice. I could only guess its in detail definition.

**Comment by phit** [comment]
@ST-DDT archive of the original page http://web.archive.org/web/20160906180457/http://www.minecraftforge.net/wiki/Core_Mod

**Comment by ST-DDT** [comment]
We already have a small explanation there. Is it true to say that it was replaced by mixins?

**Comment by ST-DDT** [comment]
As for the `SpongeEventFactory` and `BoundedComparableValue`, I will rephrase/change the sentences accordingly tommorow.

**Comment by ST-DDT** [comment]
I guess everythin is covered now. Ready for review.

**Comment by ryantheleach** [comment]
No I do not think it's fair to say that they are replaced by Mixins.

Mixins are just one method of creating a CoreMod.

**Comment by Inscrutable** [comment]
Hmmm. I think we should link to the archived page to explain CoreMods, as ["sadly core modding is still a thing"](http://www.minecraftforge.net/forum/topic/58706-regarding-minecraft-112-and-policy-changes/). Mixins can cover a lot of it, but so can any arbitrary hack (just with less compatibility).

**Comment by Inscrutable** [comment]
Alternatively, brutalising the definition from the archive into something compact:
``A CoreMod is a Forge mod that loads before Mojang classes are loaded, allowing it to modify native Minecraft code via Access Transformers, ASM class modification, or other Java bytecode manipulation.``

**Comment by ST-DDT** [comment]
Done

File: source/about/glossary.rst [hunk 5]
```diff
@@ -54,7 +54,7 @@ Ore
   The Official Sponge plugin hosting site, not ready yet. Use the Sponge Forums instead.
 Patreon
   A Service that allows for donations to projects and organizations. If you want to support Sponge, this is the place to
-  look at! `Sponge Patreon Donation Page <https://www.patreon.com/Sponge?ty=h>`__
+  look at! `Sponge Patreon Donation Page <https://www.patreon.com/Sponge>`__
```
Comment (phit) [inline source/about/glossary.rst]: may have been a value in the past, but I couldn't find any trace of it in the Patreon docs

**Pull request merged**